openedx/core/djangoapps/schedules/management/commands/__init__.py

```diff
@@ -29,12 +32,21 @@ def add_arguments(self, parser):
             '--override-recipient-email',
             help='Send all emails to this address instead of the actual recipient'
         )
-        parser.add_argument('site_domain_name')
+        parser.add_argument(
+            'site_domain_name',
+            nargs='?',
+            default=ALL_SITES
```

Comment:
What's the purpose of using `nargs='?'` and a constant string default here?

Wouldn't it be better to leave it as-is, and run on all sites if no site domain name is given?


```suggestion
```

Or perhaps, if it needs to be backward compatible, you could compare with the "all-sites" constant.

```suggestion
help="domain name for the site to use. If you wish to run this on all sites, provide 'all-sites' as the value."
```

Please also add a help text here.

Reply:
Oh wait I see, it's a positional argument. This could work then. Perhaps defaulting to None would still be cleaner though..

Please add help text to clarify the behaviour for the user. :)

Reply:
Yep, it's a positional argument, which is why I had to use `nargs='?'` to maintain backward compatibility.

I think using `None` as the default is a far cleaner approach, and I'm not sure why I used a default string in the first place. Thanks for highlighting this; I have changed it now. :+1:

Also, added help test as suggested.